src/nft/hooks/useSubscribeTransactionState.ts

```diff
@@ -0,0 +1,38 @@
+import { BagStatus, TxStateType } from 'nft/types'
+import { useEffect, useRef } from 'react'
+import shallow from 'zustand/shallow'
+
+import { useBag } from './useBag'
+import { useSendTransaction } from './useSendTransaction'
+
+export function useSubscribeTransactionState(setModalIsOpen: (isOpen: boolean) => void) {
+  const transactionState = useSendTransaction((state) => state.state)
+  const setTransactionState = useSendTransaction((state) => state.setState)
+  const transactionStateRef = useRef(transactionState)
+  const { setBagStatus, setLocked: setBagLocked } = useBag(
+    ({ setBagExpanded, setBagStatus, setLocked }) => ({
+      setBagExpanded,
+      setBagStatus,
+      setLocked,
+    }),
+    shallow
+  )
+
+  useEffect(() => {
```

Comment:
q: did you get a chance to look into removing useEffects and instead bagStatus is directly updated only from the hooks in useFetchAssets or from direct user interaction

Reply:
not yet that is going to take some refactoring of useSendTransaction which is a rabbit hole that I haven't dove into yet